core/org.wso2.carbon.base/src/test/java/org/wso2/carbon/base/CarbonBaseUtilsTest.java

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright (c) 2017, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.base;
+
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import java.lang.reflect.Field;
+import java.net.URL;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static org.testng.Assert.assertEquals;
+
+/**
+ * Test class for CarbonBaseUtil related methods with a correct sec policy.
+ */
+public class CarbonBaseUtilsTest {
+
+    @BeforeClass
+    public void setSecurityManager() {
+        URL resourceURL = CarbonBaseUtilsTest.class.getClassLoader().getResource("");
+        if (resourceURL != null) {
+            String resourcePath = resourceURL.getPath();
+            resourcePath = resourcePath + "policy-test.policy";
+            System.setProperty("java.security.policy", resourcePath);
+            System.setSecurityManager(new SecurityManager());
+        }
+    }
+
+    @AfterClass
+    public void clearSecurityManager() {
+        System.clearProperty("java.security.policy");
+    }
+
+    @Test(groups = {"org.wso2.carbon.base"})
+    public void testGetCarbonConfigDirPath() throws ClassNotFoundException, IllegalAccessException,
+            NoSuchFieldException {
+        //setting env variables for testing getCarbonConfigDirPath() in CarbonBaseUtils
+        Map<String, String> newEnv = new HashMap<>();
```

Comment:
Let's extract setting environment variable logic to a separate method or class.

Example: https://github.com/wso2/carbon-config/blob/master/components/org.wso2.carbon.config/src/test/java/org/wso2/carbon/config/utils/EnvironmentUtils.java